src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/UIntExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal static class UIntExtensions
+    {
+        internal static bool Succeeded(this uint status)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return status >= 0x80000000;
```

Comment:
Why is this different to `MsQuicStatusHelper.Succeeded`? 
And are both needed?

Reply:
This checks for any successful status code (ex success, pending, etc). Success itself is a status code as well. I'll rename to make it more clear.